src/apps/DesktopAppPage.vala

```diff
@@ -400,5 +563,470 @@ namespace Ilia {
+            
+            string action_id;
+            actions_model.@get(iter, ACTION_VIEW_COLUMN_ID, out action_id);
+            
+            // Handle separator
+            if (action_id == "generic-separator") {
+                return;
+            }
+            
+            // Handle generic actions
+            if (action_id.has_prefix("generic-")) {
+                string? desktop_file_path = app_info.get_filename();
+                string wm_cmd = get_wm_cli(this.session_controller.get_wm_name());
+                
+                switch (action_id) {
+                    case "generic-favorite":
+                        // Toggle favorite status
+                        toggle_favorite(app_info);
+                        hide_app_actions();
+                        break;
+                
+                    case "generic-terminal":
+                        try {
+                            // Get the exec line from desktop file
+                            var key_file = new KeyFile();
+                            key_file.load_from_file(desktop_file_path, KeyFileFlags.NONE);
+                            string exec_line = key_file.get_string("Desktop Entry", "Exec");
+                            
+                            // Strip % arguments
+                            string clean_exec = "";
+                            bool in_arg = false;
+                            
+                            for (int i = 0; i < exec_line.length; i++) {
+                                if (exec_line[i] == '%') {
+                                    in_arg = true;
+                                    continue;
+                                }
+                                
+                                if (in_arg) {
+                                    in_arg = false;
+                                    continue;
+                                }
+                                
+                                clean_exec += exec_line[i].to_string();
+                            }
+                            
+                            // Launch in terminal
+                            string terminal_cmd = "x-terminal-emulator -e " + clean_exec.strip();
+                            string[] args = {"sh", "-c", terminal_cmd};
+                            Process.spawn_async(null, args, null, SpawnFlags.SEARCH_PATH, null, null);
+                            hide_app_actions();
+                            session_controller.quit();
+                        } catch (Error e) {
+                            stderr.printf("Failed to launch in terminal: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        break;
+                        
+                    case "generic-shortcut":
+                        try {
+                            // Create desktop shortcut
+                            string home_dir = Environment.get_home_dir();
```

Comment:
This case is not relevant to i3 and sway since they don't have a desktop that can show icons